### PR TITLE
Add config for private submodules from external git servers

### PR DESCRIPTION
**Pull request by simeonkorchev**
This PR is related to the use case in which a given container repository hosted on a given git server has private git submodules, which are part from another git server.
The PR adds an additional optional `submodules` config section, which contains information regarding the git server that hosts the private git submodules.
The information includes the `host` of the git server, `username` and `password` for a git user, which has a permissions to read/write the contained submodule.
Related to: https://github.com/concourse/git-resource/issues/270

### CODE REVIEW

**Comment by simeonkorchev** [comment]
@vito  would you take a look at the PR?

File: README.md [hunk 1]
```diff
@@ -52,6 +52,17 @@ Tracks the commits in a [git](http://git-scm.com/) repository.
   the `branch`. Patterns are [glob(7)](http://man7.org/linux/man-pages/man7/glob.7.html)
   compatible (as in, bash compatible).
 
+* `submodules`: *Optional.* List of credentials for HTTP(s) auth when pulling/pushing private git submodules which are not stored in the same git server as the container repository.
```
Comment (vito) [inline README.md]: Can we call this `submodule_credentials` or something instead? I could see it being confused with the `submodules` param on the `get` step.

Comment (simeonkorchev) [reply README.md]: That's a good catch! I have corrected it in the latest commit

**Comment by gowrisankar22** [comment]
@vito can you check and merge the PR. We are waiting for this feature

**Pull request merged**